### PR TITLE
Case Share Component

### PR DESCRIPTION
**Pull request by ronaldmansveld**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*
*
*

### CODE REVIEW

File: projects/exui-common-lib/src/lib/models/basic-user.model.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export interface BasicUser {
+    [key: string]: any;
```
Comment (paddybasi) [inline projects/exui-common-lib/src/lib/models/basic-user.model.ts:2]: Is this needed? it opens up the interface to have any properties

File: projects/exui-common-lib/src/lib/models/shareable-case.model.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { BasicUser } from './basic-user.model';
+
+export interface ShareableCase {
+    [key: string]: any;
```
Comment (paddybasi) [inline projects/exui-common-lib/src/lib/models/shareable-case.model.ts:4]: Again is this required?

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    return this.subject.asObservable().pipe(
+      distinctUntilChanged()
+    );
+  }
+
+  constructor() { }
+
+  public setCases(orgId: string, cases: SharedCase[]): void {
+    // TODO: check organisation policy against orgId
+    parseInt(orgId, 10); // do something with orgId for now to stop linting errors
```
Comment (paddybasi) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts:25]: Not understanding why this is required? If orgId is meant to be an int then shouldn't it be typed as such?

File: projects/exui-common-lib/src/lib/components/selected-case-list/selected-case-list.component.html [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<xuilib-selected-case
+  *ngFor="let case of cases"
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case-list/selected-case-list.component.html]: added

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+export class SelectedCaseComponent implements OnInit {
+
+  @Input() public case: SharedCase;
+  @Input() public opened = false;
+
+  @Output() public unselect = new EventEmitter<SharedCase>();
+
+  constructor() { }
+
+  public ngOnInit() {
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:19]: can't be removed as this implement ngInit.

File: projects/exui-common-lib/src/lib/models/basic-user.model.ts (hunk 1 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/models/basic-user.model.ts]: not added by me, but removed now.

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (paddybasi) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:19]: So can you not just remove the ```implements onInit```

[the remark this answers is not in the document]
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:19]: I would like to leave it as this is reserved method in angular. Also if in future someone needs to add logic in init they don't need to add this code back.

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts (hunk 3 above)
Comment (paddybasi) [reply projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts:25]: orgId isn't being used, can you not refactor the method (can probably use your ide for this) to remove it?

File: projects/exui-common-lib/src/lib/components/selected-case-list/selected-case-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { SharedCase } from '../../models/case-share.model';
+
+@Component({
+  selector: 'xuilib-selected-case-list',
+  templateUrl: './selected-case-list.component.html',
+  styleUrls: ['./selected-case-list.component.scss']
+})
+export class SelectedCaseListComponent implements OnInit {
```
Comment (paddybasi) [inline projects/exui-common-lib/src/lib/components/selected-case-list/selected-case-list.component.ts:9]: So you can remove the ```implements OnInit``` here then also remove the ```public ngOnIt()``` below

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (paddybasi) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:19]: Let's not go down the route of 'if needed in future'. We should ideally be practising TDD style, which means minimal code to get the required functionality

**Pull request merged**